migration/workspace-query-separation/src/main/java/ddf/migration/workspace/query/separation/WorkspaceQueryMigration.java

```diff
@@ -0,0 +1,226 @@
+    }
+
+    LOGGER.info("Completed workspace query data migration");
+    System.out.println("Completed: Workspace Query Data Migration");
+  }
+
+  private void migrateWorkspaceMetacard(Metacard workspaceMetacard) {
+    LOGGER.info("Migrating workspace metacard with id [{}]", workspaceMetacard.getId());
+
+    final Attribute queriesAttribute = workspaceMetacard.getAttribute(WORKSPACE_QUERIES);
+
+    if (queriesAttribute == null || queriesAttribute.getValues().isEmpty()) {
+      LOGGER.info(
+          "Workspace metacard with id [{}] did not contain query data", workspaceMetacard.getId());
+      return;
+    }
+
+    final List<Serializable> queriesAsXml = queriesAttribute.getValues();
+
+    LOGGER.info(
+        "Beginning migration of {} query metacards from workspace [{}]",
+        queriesAsXml.size(),
+        workspaceMetacard.getId());
+
+    List<Metacard> queryMetacards =
+        queriesAsXml
+            .stream()
+            .filter(String.class::isInstance)
+            .map(String.class::cast)
+            .map(xml -> xml.getBytes(StandardCharsets.UTF_8))
+            .map(ByteArrayInputStream::new)
+            .map(this::xmlToMetacard)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+
+    LOGGER.info("Creating query metacards for workspace [{}]", workspaceMetacard.getId());
+
+    CreateResponse response = createMetacards(queryMetacards);
+
+    if (response != null) {
+      List<String> createdMetacardIds =
+          response.getCreatedMetacards().stream().map(Metacard::getId).collect(Collectors.toList());
+
+      LOGGER.info("Created query metacards with IDs {}", createdMetacardIds);
+
+      workspaceMetacard.setAttribute(
+          new AttributeImpl(WORKSPACE_QUERIES, (Serializable) createdMetacardIds));
+      updateMetacard(workspaceMetacard);
+
+      LOGGER.info(
+          "Updated workspace metacard [{}] with query metacard associations {}",
+          workspaceMetacard.getId(),
+          createdMetacardIds);
+    }
+  }
+
+  private UpdateResponse updateMetacard(Metacard metacard) {
+    return AccessController.doPrivileged(
+        (PrivilegedAction<UpdateResponse>)
+            () -> {
+              try {
+                return catalogFramework.update(new UpdateRequestImpl(metacard.getId(), metacard));
+              } catch (IngestException | SourceUnavailableException e) {
+                LOGGER.debug("Error updating workspace metacard [{}]", metacard.getId());
+              }
+              return null;
+            });
+  }
+
+  private CreateResponse createMetacards(List<Metacard> metacards) {
+    return AccessController.doPrivileged(
+        (PrivilegedAction<CreateResponse>)
+            () -> {
+              try {
+                return catalogFramework.create(new CreateRequestImpl(metacards));
+              } catch (IngestException | SourceUnavailableException e) {
+                LOGGER.debug("Error while create query metacards {}", metacards);
```

Comment:
Should be "creating" instead of "create"

Reply:
Consider making the error message more specific to the exception that was thrown

Reply:
I could break out each `catch` block into its own section to provide a better message however I think including the exception in the stack trace will probably provide just as much information